### PR TITLE
All easy gfg solution added

### PR DESCRIPTION
**Pull request by Vipullakum007**
## Related Issue

fixed : #386 

## Description

i have added solution for all easy problems
**i have added 10 files for solution of 10 problems**

Screenshot 2024-06-05 221647

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

Screenshot (677)

Screenshot (678)

Screenshot (679)

Screenshot (680)

Screenshot (681)

Screenshot (682)

Screenshot (683)

Screenshot (684)

Screenshot (685)

Screenshot (686)

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Vipullakum007** [comment]
hey @Ajay-Dhangar , please make it` level 3 `because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`

**Comment by Ajay-Dhangar** [comment]
> hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`

@Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.

File: dsa-solutions/gfg-solutions/0023-Find-triplet-with-zero-sum.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+### Your Task
+You don't need to read input or print anything. Your task is to complete the function `findTriplet()` which takes the integer `N` and integer array `arr[]` as parameters and returns true if there is a triplet with a sum of zero, otherwise false.
+
+**Expected Time Complexity:** O(N^2)  
+**Expected Auxiliary Space:** O(1)
+
+### Constraints
+- 1 ≤ N ≤ 10^5  
+- -10^3 ≤ arr[i] ≤ 10^3
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0023-Find-triplet-with-zero-sum.md:59]: replace 

```md
- 1 ≤ N ≤ 10^5  
- -10^3 ≤ arr[i] ≤ 10^3
```

to 

```md
- $1 ≤ N ≤ 10^5$
- $-10^3 ≤ arr[i] ≤ 10^3$
```

File: dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+### Your Task
+You only need to complete the function `minIndexChar()` that returns the index of the answer in `str` or returns `-1` if no character of `patt` is present in `str`.
+
+**Expected Time Complexity:** O(N)  
+**Expected Auxiliary Space:** O(Number of distinct characters)
+
+### Constraints
+- 1 ≤ |str|,|patt| ≤ 10^5
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md:57]: replace

```md
- 1 ≤ |str|,|patt| ≤ 10^5
- 'a' ≤ str[i], patt[i] ≤ 'z'
```

to

```md
- $1 ≤ |str|,|patt| ≤ 10^5$
- $'a' ≤ str[i], patt[i] ≤ 'z'$
```

File: dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    }
+}
+```
+
+  </TabItem>
+</Tabs>
+
+### Complexity Analysis
+
+- **Time Complexity:** O(N), where N is the length of the string `str`. We iterate through each character in `patt` and use the `find` or `indexOf` method, which runs in O(N) time.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md:172]: replace `O(N)` to `$O(N)$` and `O(1)` to `$O(1)$`

File: dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+Explanation:
+0 is connected to 1, 2.
+Starting from 0, it will go to 1 then 2, thus BFS will be 0 1 2.
+```
+
+### Your Task
+
+You don't need to read input or print anything. Your task is to complete the function `bfsOfGraph()` which takes the integer `V` denoting the number of vertices and adjacency list as input parameters and returns a list containing the BFS traversal of the graph starting from the 0th vertex from left to right.
+
+**Expected Time Complexity:** O(V + E)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:47]: replace `O(V + E)` to `$O(V + E)$` and `O(V)` to `$O(V)$`

File: dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  </TabItem>
+</Tabs>
+
+---
+
+## Complexity analysis
+
+The provided solutions efficiently perform a Breadth First Search (BFS) traversal of a directed graph. By starting from the 0th vertex and using a queue to manage the traversal, the algorithms ensure that all reachable vertices are visited in the correct order. The solutions operate in O(V + E) time and use O(V) space complexity, where V and E are the numbers of vertices and edges in the graph, respectively.
+
+**Time Complexity:** O(V + E)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:216]: replace

`O(V + E)` $=>$ `$O(V + E)$`
`O(V)` $=>$ `$O(V)$`

File: dsa-solutions/gfg-solutions/Easy problems/square-root.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+  </TabItem>
+</Tabs>
+
+## Complexity Analysis
+
+The provided solutions efficiently find the floor value of the square root of a given integer `x` using binary search. This approach ensures a time complexity of O(log N) and an auxiliary space complexity of O(1). The algorithms are designed to handle large values of `x` up to 10^7 efficiently without relying on built-in square root functions.
+
+**Time Complexity:** O(log N)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/square-root.md:192]: replace `O(log N)` to `$O(log N)$` and `O(1)` to `$O(1)$`

File: dsa-solutions/gfg-solutions/Easy problems/square-root.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+### Your Task
+
+You don't need to read input or print anything. The task is to complete the function `floorSqrt()` which takes `x` as the input parameter and returns its square root. Note: Try solving the question without using the sqrt function. The value of `x` ≥ 0.
+
+**Expected Time Complexity:** O(log N)  
+**Expected Auxiliary Space:** O(1)
+
+**Constraints**
+- 1 ≤ x ≤ 10^7
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/square-root.md:46]: replace `1 ≤ x ≤ 10^7` to `$1 ≤ x ≤ 10^7$`

File: dsa-solutions/gfg-solutions/Easy problems/square-root.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+Input: x = 4
+Output: 2
+Explanation: Since 4 is a perfect square, its square root is 2.
+```
+
+### Your Task
+
+You don't need to read input or print anything. The task is to complete the function `floorSqrt()` which takes `x` as the input parameter and returns its square root. Note: Try solving the question without using the sqrt function. The value of `x` ≥ 0.
+
+**Expected Time Complexity:** O(log N)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/square-root.md:42]: replace` O(log N)` to `$O(log N)$` and `O(1)` to `$O(1)$`

File: dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+}
+```
+
+  </TabItem>
+</Tabs>
+
+### Complexity Analysis
+
+- **Time Complexity:** O(N), where N is the number of nodes in the linked list. We traverse the entire list once.
+- **Space Complexity:** O(1), as we only use a constant amount of extra space for pointers.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:182]: replace `O(1)` to `$O(1)$` and `O(N)` to `$O(N)$`

File: dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+```
+
+### Your Task
+The task is to complete the function `reverseList()` with head reference as the only argument and should return the new head after reversing the list.
+
+**Expected Time Complexity:** O(N)  
+**Expected Auxiliary Space:** O(1)
+
+### Constraints
+1 ≤ N ≤ 10^4
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:52]: **Bad practice**

```md
### Constraints
1 ≤ N ≤ 10^4
```

**Good practice**

```md
### Constraints

- $1 ≤ N ≤ 10^4$
```

File: dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+10 9 8 7 2
+
+Explanation:
+After reversing the list, elements are 10->9->8->7->2.
+```
+
+### Your Task
+The task is to complete the function `reverseList()` with head reference as the only argument and should return the new head after reversing the list.
+
+**Expected Time Complexity:** O(N)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:48]: replace `O(N)` to `$O(N)$`
and `O(1)` to `$O(1)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now update code for all files based on suggestions

**Comment by Vipullakum007** [comment]
> > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> 
> @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.

please can you point out what you want so that i can better / i can make professionally
i have ask this question many times please answer me.

**Comment by Vipullakum007** [comment]
> > > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> > 
> > 
> > @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.
> 
> please can you point out what you want so that i can better / i can make professionally i have ask this question many times please answer me.

hey @Ajay-Dhangar , updated as per your suggestion , you can review it. please answer

**Comment by Ajay-Dhangar** [comment]
> > > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> > 
> > 
> > @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.
> 
> please can you point out what you want so that i can better / i can make professionally i have ask this question many times please answer me.

Yah sure, 
I review your work then may be assign good level if your contents is good and fully based on our structures

File: dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+### Your Task
+
+You don't need to read input or print anything. Your task is to complete the function `bfsOfGraph()` which takes the integer `V` denoting the number of vertices and adjacency list as input parameters and returns a list containing the BFS traversal of the graph starting from the 0th vertex from left to right.
+
+**Expected Time Complexity:** $ O(V + E) $ 
+**Expected Auxiliary Space:** $O(V)$
+
+**Constraints**
+- $ 1 ≤ V, E ≤ 10^4 $
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:51]: replace `- $ 1 ≤ V, E ≤ 10^4 $` to `- $1 ≤ V, E ≤ 10^4$`

File: dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+  </TabItem>
+</Tabs>
+
+---
+
+## Complexity analysis
+
+The provided solutions efficiently perform a Breadth First Search (BFS) traversal of a directed graph. By starting from the 0th vertex and using a queue to manage the traversal, the algorithms ensure that all reachable vertices are visited in the correct order. The solutions operate in O(V + E) time and use O(V) space complexity, where V and E are the numbers of vertices and edges in the graph, respectively.
+
+**Time Complexity:** $ O(V + E)  $
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:216]: not using space

and replace `$ O(V + E) $` to `$O(V + E)$`

and also, `$ O(V)$` to `$O(V)$`

File: dsa-solutions/gfg-solutions/Easy problems/check-for-balanced-tree.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+
+```
+Input: root = []
+Output: true
+```
+
+### Constraints
+
+- The number of nodes in the tree is in the range `[0, 5000]`.
+- $ -10^4 <=$ Node.val $<= 10^4 $
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/check-for-balanced-tree.md:53]: replace `- $ -10^4 <=$ Node.val $<= 10^4 $` to `- $-10^4 <=$ Node.val $<= 10^4$`

File: dsa-solutions/gfg-solutions/Easy problems/dfs-of-graph.md [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+}
+```
+
+  </TabItem>
+</Tabs>
+
+
+### Complexity Analysis
+
+- **Time Complexity:** $ O(V + E)  $
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/dfs-of-graph.md:196]: same issue

File: dsa-solutions/gfg-solutions/Easy problems/implement-two-stacks-in-an-array.md [hunk 16]
```diff
@@ -0,0 +1,349 @@
...
+    }
+}
+```
+
+  </TabItem>
+</Tabs>
+
+## Complexity Analysis
+
+The provided solutions efficiently implement two stacks in a single array using pointers. This approach ensures a time complexity of O(1) for all stack operations and an auxiliary space complexity of O(1). The algorithms are designed to handle up to 10^4 queries efficiently without relying on dynamic data structures.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Easy problems/implement-two-stacks-in-an-array.md:340]: replace `O(1)` to `$O(1)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions and can't be use space for maths expression. And review all files code and resolve issues

**Comment by Vipullakum007** [comment]
hey @Ajay-Dhangar  ,updated as per your suggestion

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**